xt_training/runner.py

```diff
@@ -165,10 +165,31 @@ def __call__(self, loader, mode=None, return_preds=False):
                     if isinstance(y, torch.Tensor):
                         y = y.to(device)
                     elif isinstance(y, Iterable):
-                        y = [y_i.to(device) for y_i in y]
-
-                y_pred = model(x)
-                loss_batch = loss_fn(y_pred, y)
+                        y_tmp = []
+                        for y_i in y:
+                            if isinstance(y_i, dict):
+                                y_i = {k: v.to(device) for k, v in y_i.items()}
+                            else:
+                                y_i.to(device)
+                            y_tmp.append(y_i)
+                        y = y_tmp
```

Comment:
Can we add a comment describing what is happening here and what the use case is?

Reply:
Comment added.